src/Workspaces/Core/Portable/FindSymbols/SyntaxTree/SyntaxTreeInfo.cs

```diff
@@ -170,10 +171,20 @@ public static Task<SyntaxTreeDeclarationInfo> GetDeclarationInfoAsync(Document d
                             // TryGetDeclaredSymbolInfo, then this will at least prevent us from returning bad spans
                             // and will prevent the crash from occurring.
                             DeclaredSymbolInfo declaredSymbolInfo;
-                            if (syntaxFacts.TryGetDeclaredSymbolInfo(node, out declaredSymbolInfo) &&
-                                root.FullSpan.Contains(declaredSymbolInfo.Span))
+                            if (syntaxFacts.TryGetDeclaredSymbolInfo(node, out declaredSymbolInfo))
                             {
-                                declaredSymbolInfos.Add(declaredSymbolInfo);
+                                if (root.FullSpan.Contains(declaredSymbolInfo.Span))
+                                {
+                                    declaredSymbolInfos.Add(declaredSymbolInfo);
+                                }
+                                else
+                                {
+                                    var message =
+$@"Invalid span in {nameof(declaredSymbolInfo)}.
+{nameof(declaredSymbolInfo.Span)} = {declaredSymbolInfo.Span}
+{nameof(root.FullSpan)} = {root.FullSpan}";
```

Comment:
Maybe :)

My understanding from before was that the crash was happening much further down hte line.  This pushes the watson much closer to the point that we can identify that something went wrong.  If this is not suffickient, we can hopefully keep pushing things back.